packages/adblocker/src/resources.ts

```diff
@@ -139,11 +202,77 @@ export default class Resources {
     return { body, contentType, dataUrl };
   }
 
+  public getScriptlet(name: string): string | undefined {
+    if (name.endsWith('.fn')) {
+      return undefined;
+    }
+
+    const scriptlet = this.scriptletsByName.get(name) || this.scriptletsByName.get(name + '.js');
+
+    if (scriptlet === undefined) {
+      this.scriptletsCache.set(name, '');
+      return undefined;
+    }
+
+    const [scriptletName] = scriptlet.names;
+    let script = this.scriptletsCache.get(scriptletName);
+
+    if (script !== undefined) {
+      if (script.length === 0) {
+        return undefined;
+      }
+      return script;
+    }
+
+    const dependencies = this.getScriptletDepenencies(scriptlet);
+    script = assembleScript(scriptlet.body, dependencies);
+
+    this.scriptletsCache.set(scriptletName, script);
+    return script;
+  }
+
+  private getScriptletDepenencies(scriptlet: Scriptlet): string[] {
+    const dependencies: Map<string, string> = new Map();
+    const queue: string[] = [...scriptlet.dependencies];
+
+    while (queue.length > 0) {
+      const dependencyName = queue.pop()!;
+      if (dependencies.has(dependencyName)) {
+        continue;
+      }
+      const dependency = this.scriptletsByName.get(dependencyName);
+      if (dependency === undefined) {
+        dependencies.set(
+          dependencyName,
+          `console.warn('@ghostery/adblocker: cannot find dependency: "${dependencyName}" for scriptlet: "${scriptlet.names[0]}"')`,
+        );
+        continue;
+      }
+      dependencies.set(dependencyName, dependency.body);
+      queue.push(...dependency.dependencies);
+    }
+
+    return Array.from(dependencies.values());
+  }
+
   public getSerializedSize(): number {
-    let estimatedSize = sizeOfASCII(this.checksum) + 2 * sizeOfByte(); // resources.size
+    let estimatedSize = sizeOfASCII(this.checksum) + 2; // resources.size
+
+    this.resources.forEach(({ names, body: content, contentType }) => {
```

Comment:
Nit. We use `forEach` here but `for .. of` in other parts of the file, maybe we can unify and use `for .. of` everywhere? (Same for other `forEach` below):
```suggestion
    for (const { names, body: content, contentType } of this.resources) {
```